src/main/python/deep_qa/layers/noisy_or.py

```diff
@@ -0,0 +1,89 @@
+from keras import backend as K
+from keras.layers import Layer
+from keras.constraints import Constraint
+from keras.regularizers import L1L2Regularizer
+from overrides import overrides
+
+class BetweenZeroAndOne(Constraint):
+    """
+    Constrains the weights to be between zero and one
+    """
+    def __call__(self, p):
+        # Clip values less than or equal to zero to epsilon
+        p *= K.cast(p >= 0., K.floatx())
+        leaky_zeros_mask = K.epsilon() * K.cast(K.equal(p, K.zeros_like(p)), 'float32')
+        p = p + leaky_zeros_mask
+        # Clip values greater to 1 to 1
+        p *= K.cast(p <= 1., K.floatx())
+        leaky_ones_mask = K.cast(K.equal(p, K.zeros_like(p)), K.floatx())
+        p += leaky_ones_mask
+        return p
+
+
+class NoisyOr(Layer):
+    """
+    Calculate the noisy-or probability across a given tensor axis based on
+    noisy-or equation:
+        p(x) = 1 - Prod_(i=1:N)[1 - q * p(x|y_n)]
+    where q is the noise parameter
+
+    Parameters
```

Comment:
This is nice.  Thanks for the good documentation.  One quibble, though - can you make it match the formatting used [here](https://github.com/allenai/deep_qa/blob/ba567499184b27e6f7c1747360ab86b54c5e8277/src/main/python/deep_qa/layers/attention/gated_attention.py#L33-L52)?  That will make it play nicely with our auto-generated docs.  Also, can you ask Nelson how to add this layer to the docs?  (Or follow what you see [here](https://github.com/allenai/deep_qa/pull/191/files), and ask Nelson if you need help.  He can point you to how to see the docs generated from your PR, to make sure they're what you expect.)